### PR TITLE
feat: add optional job summary-title parameter

### LINKED ISSUE #1050: Make table title configurable or based on step name
On one of my workflows, I run component and afterward E2E tests. The summary produces two tables with the identical heading "Cypress Results". I'd like to have the step name, a custom specified name or maybe just the hint if those are E2E/component tests in the table headers.

### PR DESCRIPTION
**Pull request by MikeMcC399**
- closes #1050

This PR adds an optional parameter `summary-title` to the `cypress-io/github-action`.

If specified, then the `summary-title` replaces the default Job Summary title "Cypress Results". The title is displayed at the top of a job summary.

This is especially useful when `cypress-io/github-action` is called in more than one step in a GitHub Actions workflow, so that each job summary can be easily identified with individual summary titles showing which step it came from.

## example-chrome

`summary-title` is applied to the [example-chrome](https://github.com/cypress-io/github-action/actions/workflows/example-chrome.yml) workflow.

**BEFORE**

example-chrome_before

**AFTER**

example-chrome_after

`summary-title` is here defined as "Chrome headless" in the first instance and "Chrome headed" in the second instance.

[example-chrome](https://github.com/cypress-io/github-action/actions/workflows/example-chrome.yml) is the only example workflow in this repo with more than one call of `cypress-io/github-action` in a single job.

## Documentation

- The parameter `summary-title` is added to the [README](https://github.com/cypress-io/github-action/blob/master/README.md) and some inconsistencies regarding the description of the parameter `publish-summary` are removed.

## Verification

- View the log of [example-chrome](https://github.com/cypress-io/github-action/actions/workflows/example-chrome.yml) and confirm that the Job Summary uses the customized headings "Chrome headless" in the first instance and "Chrome headed" in the second instance
- View the log of [example-basic](https://github.com/cypress-io/github-action/actions/workflows/example-basic.yml) and confirm that the Job Summary heading shows "Cypress Results"

## References

- [Supercharging GitHub Actions with Job Summaries](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/)

### CODE REVIEW

**Comment by MikeMcC399** [comment]
- Review new README section at https://github.com/MikeMcC399/github-action/tree/summary-title#job-summary-title
- Example chrome job at https://github.com/cypress-io/github-action/actions/runs/6524966440
- Example basic job at https://github.com/cypress-io/github-action/actions/runs/6524966454

**Comment by MikeMcC399** [comment]
Unrelated to changes in this PR there is currently a markdown-link-check failure:

```text
  [✖] https://www.microsoft.com/windows/ → Status: 0 Error: socket hang up
    at connResetException (node:internal/errors:720:14)
    at TLSSocket.socketCloseListener (node:_http_client:474:25)
    at TLSSocket.emit (node:events:529:35)
    at node:net:350:12
    at TCP.done (node:_tls_wrap:614:7) {
  code: 'ECONNRESET'
}
```

**Comment by alex1701c** [comment]
Thanks for the PR! This perfectly fits the need I described in the issue :)  

The code looks also good and the change is well documented.

**Comment by MikeMcC399** [comment]
@alex1701c 

> Thanks for the PR! This perfectly fits the need I described in the issue :)
> 
> The code looks also good and the change is well documented.

Thanks for the positive comments and your confirmation! Now it's over to the Cypress team to see if they agree to merge it.

**Comment by MikeMcC399** [comment]
@mschile 

Is there any chance you could also look at this enhancement PR?

File: README.md [hunk 1]
```diff
@@ -54,7 +54,8 @@
 - Use [timeouts](#timeouts) to avoid hanging CI jobs
 - Print [Cypress info](#print-cypress-info) like detected browsers
 - Run [tests nightly](#nightly-tests) or on any schedule
-- Suppress [test summary](#suppress-test-summary)
+- Specify [job summary title](#job-summary-title)
```
Comment (emilyrohrbough) [inline README.md:57]: I like this change. I do find the number of examples is quite overwhelming and identifying all of the supported configurations is really difficult to quickly identify in our current read me. Something that's been on my mind when reviewing this repo in general.

**Review by emilyrohrbough — APPROVED** [review]

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**